core/src/core/pmmr/pmmr.rs

```diff
@@ -396,6 +372,63 @@ where
 	}
 }
 
+impl<'a, T, B> ReadablePMMR for PMMR<'a, T, B>
+where
+	T: PMMRable,
+	B: 'a + Backend<T>,
+{
+	type Item = T::E;
+
+	fn get_hash(&self, pos: u64) -> Option<Hash> {
```

Comment:
Actually - can `get_hash()` and `get_data()` (and similar) not also be moved to the trait if we had a `backend()` fn?

Reply:
Yes, but that does mean we give access to the backend outside of the module. It might give use the temptation to accidentally use the backend directly.

Reply:
Hm on further thought, we could maybe add the `backend()` function to a separate trait that isn't visible outside of the module..

Reply:
Turns out this is not really possible, as it would expose the private trait:
```
error[E0445]: restricted trait `core::pmmr::pmmr::PMMRBackend<B>` in public interface
```
So I'd rather keep it like it is, with some duplicate code.